### ISSUE TITLE
Rule `ARG001` triggers for pytest fixtures that depend on other fixtures

### ISSUE THREAD

**Issue by cbornet** [astral-sh/ruff#24034]
### Summary

Rule `ARG001` triggers for pytest fixtures that depend on other fixtures.

```python
@pytest.fixture
def foo():
    print("before foo")
    yield
    print("after foo")


@pytest.fixture
def bar(foo):  # ARG001 Unused function argument: `foo`
    print("bar fixture activated")
```

It would be nice to be able to exclude this situation from `ARG` violations.

Notes:
* `@pytest.mark.usefixtures` works on test methods, not on fixtures so it can't be used here.
* Renaming `foo` to `_foo` would make clear that the arg is unused but would also make `_foo` as a fixture look like it's private which may not be the case.
* Another work-around is to use `request.getfixturevalue("foo")` but that's quite ugly and doesn't expose well the dependency graph.

**Comment by ntBre** [comment]
Thanks for the report! This was previously discussed a bit in https://github.com/astral-sh/ruff/issues/4673#issuecomment-1789036069, but that issue is now closed. I think this also relates a bit to https://github.com/astral-sh/ruff/issues/18119. I think we still don't have a great workaround here.

**Comment by paveldikov** [comment]
Why not `del foo`?
